### PR TITLE
Enable strict Typescript and fix all errors

### PR DESCRIPTION
**Pull request by suevalov**
# Purpose of PR

* Enables strict Typescript and fix all errors
* Removes `.css.d.ts` files because they are generated in incorrect way
* Fixes https://github.com/contentful/forma-36/issues/137

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Comment by m10l** [comment]
@suevalov Manually checked each story in Storybook - didn't notice any issues, I'll review the code shortly.

File: packages/forma-36-react-components/src/components/Typography/Typography/Typography.tsx [hunk 1]
```diff
@@ -3,36 +3,38 @@ import cn from 'classnames';
...
+  sectionHeading: { spacing: 'l' },
+  subheading: { spacing: 'm' },
+};
+
+const defaultProps = {
+  testId: 'cf-ui-text-container',
+};
 
-export const TypographyContext = React.createContext({});
+export const TypographyContext = React.createContext(defaultConfiguration);
```
Comment (m10l) [inline packages/forma-36-react-components/src/components/Typography/Typography/Typography.tsx]: @suevalov this doesn't work as expected - it'll mean the typography components will _always_ have spacing applied.

https://deploy-preview-129--forma-36-react-components.netlify.com/?path=/story/components-typography-displaytext--default

In this Story there should be no margins applied

Comment (suevalov) [reply packages/forma-36-react-components/src/components/Typography/Typography/Typography.tsx]: Good catch, @m10l! I pushed the fix.

Comment (m10l) [reply packages/forma-36-react-components/src/components/Typography/Typography/Typography.tsx]: Awesome

File: packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx [hunk 2]
```diff
@@ -43,7 +38,7 @@ export class DropdownListItem extends Component<DropdownListItemProps> {
           className={styles['DropdownListItem__toggle-button']}
           onClick={this.props.onClick}
           onMouseEnter={this.props.onEnter}
-          onFocus={this.props.onEnter}
+          onFocus={this.props.onFocus}
```
[the remark this answers is not in the document]
Comment (suevalov) [reply packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx]: @wichniowski So, it wasn't a bug?

File: packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx [hunk 3]
```diff
@@ -17,15 +15,21 @@ export interface ControlledInputPropTypes {
...
-  };
+const defaultProps = {
+  testId: 'cf-ui-controlled-input',
+  required: false,
+  disabled: false,
+  type: 'checkbox',
+};
+
+export class ControlledInput extends Component<
+  ControlledInputPropTypes & typeof defaultProps
```
[the remark this answers is not in the document]
Comment (suevalov) [reply packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx]: Actually, `ControlledInputPropTypes` is being used as interface for other components that they extend from. So, I has to stay as it now

File: packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (wichniowski) [reply packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx]: nope

File: packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (wichniowski) [reply packages/forma-36-react-components/src/components/ControlledInput/ControlledInput.tsx]: ahhh ok cool!

File: packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (suevalov) [reply packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx]: hmm... I don't understand. Previously `onFocus` property wasn't used at all.

[the remark this answers is not in the document]
Comment (wichniowski) [reply packages/forma-36-react-components/src/components/Dropdown/DropdownListItem/DropdownListItem.tsx]: true.. what about onFocus={e => {this.props.onEnter(e); this.props.onFocus(e);}}

File: packages/forma-36-react-components/src/components/Card/AssetCard/AssetCard.tsx [hunk 4]
```diff
@@ -56,25 +57,25 @@ export class AssetCard extends Component<AssetCardProps, AssetCardState> {
         </button>
       }
     >
-      {dropdownListElements.props.children &&
```
Comment (wichniowski) [inline packages/forma-36-react-components/src/components/Card/AssetCard/AssetCard.tsx]: why did you remove that @suevalov
